Core/include/Acts/Seeding/CandidatesForMiddleSp.ipp

```diff
@@ -0,0 +1,277 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+namespace Acts {
+
+template <typename external_space_point_t>
+inline void CandidatesForMiddleSp<external_space_point_t>::setMaxElements(
+    std::size_t n_low, std::size_t n_high) {
+  m_max_size_high = n_high;
+  m_max_size_low = n_low;
+
+  // protection against default numbers
+  // it may cause std::bad_alloc if we don't protect
+  if (n_high == std::numeric_limits<std::size_t>::max() or
+      n_low == std::numeric_limits<std::size_t>::max()) {
+    return;
+  }
+
+  // Reserve enough memory for all collections
+  m_storage.reserve(n_low + n_high);
+  m_storage_high.reserve(n_high);
+  m_storage_low.reserve(n_low);
+}
+
+template <typename external_space_point_t>
+inline void CandidatesForMiddleSp<external_space_point_t>::pop(
+    std::vector<std::size_t>& storage, std::size_t& current_size) {
+  // Remove the candidate with the lowest weight in the collection
+  // By construction, this element is always the first element in the
+  // collection.
+  // The removal works this way: the first element of the collection
+  // is overwritten with the last element of the collection.
+  // The number of stored elements (current_size) is lowered by 1
+  // The new first element is moved down in the tree to its correct position
+  std::swap(storage[0], storage[current_size - 1]);
```

Comment:
With c++20 we would use here shift_left I guess.

Reply:
Anyways, it ma be clearer to write `std::swap(storage.front(), storage.back());`

Reply:
I'm afraid not, it is not guaranteed that `storage.back()` corresponds to the element we want to swap. This happens if we pop more then once without adding (i.e. in the `storage()` method).